src/connectors/configure/connectConfigure.js

```diff
@@ -91,13 +91,18 @@ export default function connectConfigure(renderFn = noop, unmountFn = noop) {
       },
 
       removeSearchParameters(state) {
-        // widgetParams are assumed 'controlled',
-        // so they override whatever other widgets give the state
-        return state.mutateMe(mutableState => {
-          Object.keys(widgetParams.searchParameters).forEach(key => {
-            delete mutableState[key];
-          });
-        });
+        // We leverage the Helper internals to remove the `widgetParams` from
+        // the state. The function `setQueryParameters` omit (or remove) the
```

Comment:
```suggestion
        // the state. The function `setQueryParameters` omits the
```

This wording is slightly clearer to me

Reply:
https://github.com/algolia/instantsearch.js/pull/3792/commits/bc7c101f0b228cf2a6bb92e2485679567a238189